pyfarm/master/api/jobgroups.py

```diff
@@ -0,0 +1,355 @@
+                    "user": "testuser",
+                    "main_jobtype": "Test JobType",
+                    "jobs": [],
+                    "title": "Test Group"
+                }
+
+        :statuscode 200: no error
+        :statuscode 404: the requested job group was not found
+        """
+        jobgroup = JobGroup.query.filter_by(id=group_id).first()
+
+        if not jobgroup:
+            return (jsonify(error="Requested job group %s not found" % group_id),
+                    NOT_FOUND)
+
+        jobgroup_data = jobgroup.to_dict()
+        del jobgroup_data["user_id"]
+        del jobgroup_data["main_jobtype_id"]
+
+        return jsonify(jobgroup_data), OK
+
+    def post(self, group_id):
+        """
+        A ``POST`` to this endpoint will update the specified group with the data
+        in the request.  Columns not specified in the request will be left as
+        they are.
+
+        .. http:post:: /api/v1/jobgroups/<int:id> HTTP/1.1
+
+            **Request**
+
+            .. sourcecode:: http
+
+                POST /api/v1/jobgroups/2 HTTP/1.1
+                Accept: application/json
+
+                {
+                    "user": "testuser2"
+                }
+
+            **Response**
+
+            .. sourcecode:: http
+
+                HTTP/1.1 201 OK
+                Content-Type: application/json
+
+                {
+                    "id": 2,
+                    "user": "testuser2",
+                    "main_jobtype": "Test JobType",
+                    "jobs": [],
+                    "title": "Test Group"
+                }
+
+        :statuscode 200: the job group was updated
+        :statuscode 400: there was something wrong with the request (such as
+                         invalid columns being included)
+        """
+        jobgroup = JobGroup.query.filter_by(id=group_id).first()
+
+        if not jobgroup:
+            return (jsonify(error="Requested job group %s not found" % group_id),
+                    NOT_FOUND)
+
+        if "title" in g.json:
+            jobgroup.title = g.json.pop("title")
+        if "user" in g.json:
+            username = g.json.pop("user")
+            user = User.query.filter_by(username=username).first()
+            if not user:
+                return (jsonify(error="User %s not found" % username), NOT_FOUND)
+            jobgroup.user = user
+        if "main_jobtype" in g.json:
+            jobtype_name = g.json.pop("main_jobtype")
+            jobtype = JobType.query.filter_by(name=jobtype_name).first()
+            if not jobtype:
+                return (jsonify(error="Jobtpye %s not found" % jobtype_name),
+                        NOT_FOUND)
+            jobgroup.main_jobtype = jobtype
+
+        if g.json:
+            return jsonify(error="Unkown columns: %s" % g.json), BAD_REQUEST
+
+        db.session.add(jobgroup)
+        db.session.commit()
+
+        jobgroup_data = jobgroup.to_dict()
+        del jobgroup_data["user_id"]
```

Comment:
Same function but won't cause the whole request to fail if `jobgroup_data` does not contain the keys we're trying to remove:

```python
jobgroup_data.pop("user_id', None)
jobgroup_data.pop("main_jobtype_id", None)
```